snips-nlu-ontology-parsers/src/builtin_entity_parser.rs

```diff
@@ -26,31 +25,19 @@ lazy_static! {
 }
 
 impl BuiltinEntityParser {
-    pub fn get(lang: Language) -> Arc<BuiltinEntityParser> {
-        lazy_static! {
-            static ref CACHED_PARSERS: Mutex<HashMap<String, Arc<BuiltinEntityParser>>> =
-                Mutex::new(HashMap::new());
-        }
-
-        CACHED_PARSERS
-            .lock()
-            .unwrap()
-            .entry(lang.to_string())
-            .or_insert_with(|| {
-                let supported_entity_kinds = BuiltinEntityKind::supported_entity_kinds(lang);
-                let ordered_entity_kinds = OutputKind::all()
-                    .iter()
-                    .map(|output_kind| output_kind.into_builtin())
-                    .filter(|builtin_entity_kind| supported_entity_kinds.contains(&builtin_entity_kind))
-                    .collect();
+    pub fn new(lang: Language) -> Self {
+        let supported_entity_kinds = BuiltinEntityKind::supported_entity_kinds(lang);
+        let ordered_entity_kinds = OutputKind::all()
+            .iter()
+            .map(|output_kind| output_kind.into_builtin())
+            .filter(|builtin_entity_kind| supported_entity_kinds.contains(&builtin_entity_kind))
+            .collect();
 
-                Arc::new(BuiltinEntityParser {
-                    parser: build_parser(lang.into_builtin()).unwrap(),
-                    lang,
-                    supported_entity_kinds: ordered_entity_kinds,
-                })
-            })
-            .clone()
+        BuiltinEntityParser {
+            parser: build_parser(lang.into_builtin()).unwrap(),
```

Comment:
We could maybe get rid of this `unwrap` ?

Reply:
Ok I see, on the nlu part we're protected by the enum of rustling, but rustling may fail to initialize if it doesn't have its internal resource. I agree that, in this configuration, this is more up to rustling to protect itself against a potential lack of resources.

But as we're going to extract every resources related to language from binaries, we should be prepared, at any point in the chain, having an error related to resources... That's why I still think we should handle this `Result`. WDYT?

Reply:
Not sure about that:
- we know what it will cost us if we do that (painful `Result` climbing up in the call stack)
- we don't know what we gain in the process: we have not idea yet about how we are going to package languages, and we should *anticipate for change* which in this case translates to *anticipate for uncertainty* in my opinion.

Reply:
Having a stack of `Result` isn't so painful in IMO (in ffi maybe) but I tend to agree that the resources problem is a problem to solve in the future (but not so far ^^).
Let's merge this now but I pretty sure we'll come to this unwrap at some point 😄

Reply:
:godmode:

Reply:
😬